### ISSUE TITLE
PBR Materials Support

### ISSUE THREAD

**Issue by tonyf** [rnd-team-dev/plotoptix#49]
Do you have any examples on how to use PBR materials in plotoptix? I'm working on transitioning rendering a trimesh scene from pyrender to plotoptix and trying to figure out the easiest way to map materials

**Comment by robertsulej** [comment]
There is a [sample notebook](https://github.com/rnd-team-dev/plotoptix/blob/master/examples/1_basics/4_materials_configuration.ipynb); cell 4. has a few examples on how to use `make_material()` function. Function is documented [here](https://plotoptix.rnd.team/materials.html#plotoptix.materials.make_material), and [here](https://plotoptix.rnd.team/enums.html#plotoptix.enums.MaterialType) are available shaders.

For a better performance there are separate shaders for each basic case of materials, so e.g. transparency is handled only with `Transparent...` shaders.

Other not obvious parameters are:
- RGB colors in `Transmissive` (glass, etc) materials are interpreted as the attenuation length. The range is 0 to inf., e.g. red glass color would be [100, 0, 0] and the actual color depends on the size of object, just like in the real world.
- Roughness is a parameter of Oren-Nayar model, small values work well (~0.01). Base roughness can be modulated with the texture values.

Let me know if you need to setup something specific.

**Comment by skiler07** [comment]
@robertsulej A little more on the PBR materials...

There is a [Disney paper,](https://[media.disneyanimation.com/uploads/production/publication_asset/48/asset/s2012_pbs_disney_brdf_notes_v3.pdf](https://media.disneyanimation.com/uploads/production/publication_asset/48/asset/s2012_pbs_disney_brdf_notes_v3.pdf)) which somewhat defines the standard for a PBR material. You can find the defined parameters in this paper. 

For PlotOptiX, it seems that we need to specify the `MaterialType` to create a new material which will have a unique Radiance and Occlusion programs. 

Now, in PBR, I do not need a new Occlusion program, in order to create a glass or metal. I achieve that material by tweaking metalness, roughness and other parameters. I understand having a custom shader for non PBR materials e.g. a ShadowCatcher is a custom material. 

So I'm curious if it's possible to add `MetrialType="PBR"`, which will take parameters as defined in the Disney paper above.

**Comment by robertsulej** [comment]
Ah.. there was a message from @skiler07 that I didn't manage to answer.

Yes, materials setup is still ugly in plotoptix. And right, now it is not possible to use e.g. metalness texture on a glass. These two have different radiance program codes, which makes them faster but less universal.

I'll add something more standard. Now I have some work in queue, but I'll arrive to this shader at some point as well. :)

**Comment by robertsulej** [comment]
Yes, that is on the list. I'll work on all the materials related issues after the noise-balanced work distribution and support for the micromesh displacement. Work distribution is practicaly ready, then displacement can take ~2 weekends :)
